### DISCUSSION TITLE
Unsharp mask using bilateral blur

### DISCUSSION THREAD

**Discussion by DanielDrs** [ImageMagick/ImageMagick#5507] [Development]
Hallo,

a usual unsharp mask works like     sharpened_image  :=  original_image + factor * (original_image - blurred_image)
whereas blurred_image is computed by applying a gaussian blur to the original image.

I could imagine to use a bilateral blur instead of a gaussian blur to prevent an oversharpening of already sharp edges.  
It seems to be a quite straightforward idea, so I wonder whether it is already implemented?

Best regards, Daniel

**Comment by fmw42** [comment]
bilateral filter is already implemented in Imagemagick 7.  See https://imagemagick.org/script/command-line-options.php#bilateral-blur

**Comment by snibgo** [comment]
IM does not contain an operation that does an unsharp with bilateral blur. You can easily do it yourself by blurring a clone, and compositing that with the original with `-compose blend` or `-compose mathematics`. See, for example, [Adaptive blur and sharpen](http://im.snibgo.com/adapblrshp.htm).

**Reply by DanielDrs** [reply]
OK,  thanks.
